tui: dump termcap info if 'verbose' >= 3

justinmk opened the pull request:
With this PR, users can get terminal debugging info by starting Nvim with 'verbose' level 3:

    nvim -V3log

(Info will be flushed to the "log" file on exit.)

Note: this is like Vim's `:set termcap`, which was removed in Nvim
(and would be very awkward to restore because of UI decoupling).

justinmk committed:
tui: dump termcap info if -V3 ('verbose' >= 3)

Get terminal debugging info by starting Nvim with 'verbose' level 3:

    nvim -V3log

This is like Vim's `:set termcap`, which was removed in Nvim (and would
be very awkward to restore because of the decoupled UI).

justinmk committed:
str2specialbuf(): fix comparison

regression by 832c158a663c

justinmk committed:
msg_outtrans_special(): `const` some strings

justinmk committed:
tui: set descriptions on termcap extensions

jamessan commented:
I was also thinking that `:checkhealth` should include `:set term?` in the terminal section.

justinmk commented:
@jamessan `:set term?` doesn't work, only `:echo &term` works, but yeah.

justinmk committed:
msg_outtrans_special(): handle NULL

This is convenient for terminfo_info_msg().

justinmk committed:
test: tui_spec: narrower scope for timeout tweaks

justinmk committed:
test: macOS 10.13: unibilium cannot find "xterm" terminfo

On some macOS versions we can't find the terminfo for whatever reason,
so just skip the test if it fails.

justinmk committed:
test: retry(): fix time calculation

libuv caches the results of uv.now() until the next loop tick. If a test
does not spin the libuv event loop, retry() enters an infinite cycle.

justinmk committed:
test: write_file(): support append-mode

justinmk committed:
test/tui: -V3log logs terminfo values

justinmk committed:
tui: flush -V3 ('verbose' >= 3) info ASAP
